sbi/inference/snle/mnle.py

```diff
@@ -0,0 +1,239 @@
+                be called with the first batch of simulations (theta, x), which can
+                thus be used for shape inference and potentially for z-scoring. It
+                needs to return a PyTorch `nn.Module` implementing the density
+                estimator. The density estimator needs to provide the methods
+                `.log_prob` and `.sample()`.
+            device: Training device, e.g., "cpu", "cuda" or "cuda:{0, 1, ...}".
+            logging_level: Minimum severity of messages to log. One of the strings
+                INFO, WARNING, DEBUG, ERROR and CRITICAL.
+            summary_writer: A tensorboard `SummaryWriter` to control, among others, log
+                file location (default is `<current working directory>/logs`.)
+            show_progress_bars: Whether to show a progressbar during simulation and
+                sampling.
+            unused_args: Absorbs additional arguments. No entries will be used. If it
+                is not empty, we warn. In future versions, when the new interface of
+                0.14.0 is more mature, we will remove this argument.
+        """
+
+        if isinstance(density_estimator, str):
+            assert (
+                density_estimator == "mnle"
+            ), f"""MNLE can be used with preconfigured 'mnle' density estimator only, 
+                not with {density_estimator}."""
+        kwargs = del_entries(locals(), entries=("self", "__class__", "unused_args"))
+        super().__init__(**kwargs, **unused_args)
+
+    def append_simulations(
+        self,
+        theta: Tensor,
+        x: Tensor,
+        from_round: int = 0,
+    ) -> "LikelihoodEstimator":
+        r"""Store parameters and simulation outputs to use them for later training.
+
+        Data are stored as entries in lists for each type of variable (parameter/data).
+
+        Stores $\theta$, $x$, prior_masks (indicating if simulations are coming from the
+        prior or not) and an index indicating which round the batch of simulations came
+        from.
+
+        Args:
+            theta: Parameter sets.
+            x: Simulation outputs.
+            from_round: Which round the data stemmed from. Round 0 means from the prior.
+                With default settings, this is not used at all for `SNLE`. Only when
+                the user later on requests `.train(discard_prior_samples=True)`, we
+                use these indices to find which training data stemmed from the prior.
+
+        Returns:
+            NeuralInference object (returned so that this function is chainable).
+        """
+
+        theta, x = validate_theta_and_x(theta, x, training_device=self._device)
+        # Test net here to avoid copying the train function from base.
+        assert isinstance(
+            self._build_neural_net(theta, x), MixedDensityEstimator
+        ), """Invalid density estimtor for MNLE, pass 'mnle' or pass a built function
+        similar to sbi.neural_nets.mnle.build_mnle"""
```

Comment:
So, if we remove this assert, we can remove the entire `append_simulations` here, no?

I'd prefer removing it then. Also, if the user creates their own mixed density estimator (and call the class anything other than MixedDensityEstimator), this will fail I guess.

Reply:
yes, good point. and maybe `pyright` will even help us preventing users from passing something else here.